ibis/backends/bigquery/__init__.py

```diff
@@ -171,10 +172,14 @@ def _session_dataset(self):
     def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
         raw_name = op.name
 
-        project = self._session_dataset.project
-        dataset = self._session_dataset.dataset_id
+        session_dataset = self._session_dataset
+        project = session_dataset.project
+        dataset = session_dataset.dataset_id
 
-        if raw_name not in self.list_tables(database=(project, dataset)):
+        table_ref = bq.TableReference(session_dataset, raw_name)
+        try:
+            self.client.get_table(table_ref)
```

Comment:
Small nit - this reads clearer to me since `table_ref` was never reused.

```suggestion
        try:
            self.client.get_table(bq.TableReference(session_dataset, raw_name))
```